### PR TITLE
fix support for WebWorkers

### PR DESCRIPTION
**Pull request by dryajov**
`window` is not defined inside WebWorkers/ServiceWorkers, fallback to `self` instead.

### CODE REVIEW

File: src/crypto-browser.js [hunk 1]
```diff
@@ -6,14 +8,13 @@ const nodeify = require('nodeify')
 const webCrypto = getWebCrypto()
 
 function getWebCrypto () {
```
Comment (dryajov) [inline src/crypto-browser.js]: Any reason why `getWebCrypto` is defined twice in separate files? Should it be moved to some utils lib/file?

Comment (dignifiedquire) [inline src/crypto-browser.js]: Yeah if you could do this please. I think I forgot to clean this up when the second one was introduced.

File: src/crypto-sha1-2-browser.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
+/* global self */
+
 'use strict'
 
 const nodeify = require('nodeify')
 
 const webCrypto = getWebCrypto()
 
 function getWebCrypto () {
```
Comment (dryajov) [inline src/crypto-sha1-2-browser.js]: This file (`crypto-sha1-2-browser.js`) adds a new method `sha3` that is not present in `crypto-browser.js`, other than that they are exactly the same, should we remove `crypto-browser.js`?

**Comment by kumavis** [comment]
for browserify (not sure about webpack) you can use the keyword `global` which will be `window` or `self`

**Comment by dryajov** [comment]
Doesn't seem to be defined in webpack, can someone confirm if it supports this use case? I agree, it would be much cleaner to have the bundler handle that rather than doing it in code.

**Comment by dryajov** [comment]
@kumavis @dignifiedquire as an alternative, we can simply use `self` everywhere where modifying `window` is not required (since self is read only)? I'd hate to have the check littering the code everywhere...

**Comment by dignifiedquire** [comment]
Thank you closing in favor of #14

**Pull request closed without merging**